### PR TITLE
Added guide to additional configurations and resources for notebook servers

### PR DESCRIPTION
**Pull request by sarahmaddox**
Documents the **Configurations** and **Extra Resources** fields in the Notebooks setup screen

Fixes https://github.com/kubeflow/website/issues/1268

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1419)

### CODE REVIEW

**Comment by sarahmaddox** [comment]
/assign @lluunn 

@lluunn Please would you review this PR to check I've understood the tech correctly, and that we're giving users the right amount of information so that they can fill in the fields on the UI if they need to.

**Comment by sarahmaddox** [comment]
This PR is ready for review. Applying a hold to allow for multiple reviewers. CCing @alecglassford for tech writer review.

/hold
/cc @alecglassford

File: content/docs/notebooks/setup.md [hunk 1]
```diff
@@ -126,25 +126,65 @@ When Kubeflow is running, you can access the Kubeflow user interface (UI). If th
...
+
+1. *(Optional)* Change the setting for **enable shared memory**. The default is
+  that shared memory is enabled. Some libraries like PyTorch use shared memory
+  for multiprocessing. Currently there is no implementation in Kubernetes to 
+  activate shared memory. As a workaround, Kubeflow creates ab empty directory 
+  at `/dev/shm`. 
+
+1. *(Optional)* Specify one or more **extra resources** as a JSON string. The
+   JSON string must specify the value for one or more of the
+   `spec.containers[].resources.limits` options described in the [Kubernetes
```
Comment (alecglassford) [inline content/docs/notebooks/setup.md]: The linked page seems to describe three options for the type of resource: `cpu`, `memory`, and `hugepages-<size>`. But the example you have added below seems to reserve GPU resources (`nvidia.com/gpu`) which doesn't match any of these (as far as I can tell). So I'm a bit confused as to what I can use in this section.

Also, let's say I do want to allocate extra cpu or memory in the **Extra Resources** section.  It seems like I already allocated these resources in steps 8 and 9 of this procedure, so what is the point or difference of doing that here?

File: content/docs/notebooks/setup.md [hunk 2]
```diff
@@ -126,25 +126,65 @@ When Kubeflow is running, you can access the Kubeflow user interface (UI). If th
...
+  Kubeflow matches the labels in the **configurations** field against
+  the properties specified in the PodDefault manifest. Kubeflow then injects 
+  these configurations into all the notebook Pods on this notebook server.
+  
+    For example, enter the label `addgcpsecret` in the **configurations** field
+    to match to a PodDefault manifest containing the following configuration:
+
+    ```
+    matchLabels:
+    addgcpsecret: "true"
```
[the remark this answers is not in the document]
Comment (sarahmaddox) [reply content/docs/notebooks/setup.md]: Thanks @alecglassford! Good pickup on the indentation - I made a copy/paste error. I'll fix this in my next commit.

+1 to us needing more info about the PodDefault manifest. When I was putting this PR together, I also created issue https://github.com/kubeflow/website/issues/1423 saying we need to document the use of PodDefault. In this current PR, I'm hoping to give expert users enough of a pointer about what goes into the UI field. @lluunn I'd be glad of your input here.

File: content/docs/notebooks/setup.md [hunk 3]
```diff
@@ -126,25 +126,65 @@ When Kubeflow is running, you can access the Kubeflow user interface (UI). If th
...
+    ```
+
+    For indepth information on PodDefault usage, see the [admission-webhook
+    README](https://github.com/kubeflow/kubeflow/blob/master/components/admission-webhook/README.md).
+
+1. *(Optional)* Change the setting for **enable shared memory**. The default is
+  that shared memory is enabled. Some libraries like PyTorch use shared memory
+  for multiprocessing. Currently there is no implementation in Kubernetes to 
+  activate shared memory. As a workaround, Kubeflow creates ab empty directory
```
[the remark this answers is not in the document]
Comment (sarahmaddox) [reply content/docs/notebooks/setup.md]: Thanks!

File: content/docs/notebooks/setup.md (hunk 1 above)
Comment (sarahmaddox) [reply content/docs/notebooks/setup.md]: @lluunn Do you have an answer to Alec's question about how the JSON string `{"nvidia.com/gpu": 2}` fits into the `spec.containers[].resources.limits options`?

Comment (lluunn) [reply content/docs/notebooks/setup.md]: This should be for GPU as cpu/memory can be set in previous steps as pointed out.
We can link to this page: https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/
There is an example usage of `nvidia.com/gpu: 1`

File: content/docs/notebooks/setup.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sarahmaddox) [reply content/docs/notebooks/setup.md]: I'll go with what we have for now, as the config option is an advanced one. We'll add a link to the new doc when issue #1423 is addressed.

File: content/docs/notebooks/setup.md (hunk 1 above)
Comment (sarahmaddox) [reply content/docs/notebooks/setup.md]: Thanks both! I've clarified the section.

**Comment by sarahmaddox** [comment]
Thanks @lluunn PTAL

**Pull request merged**